New/updated development process documents.

tjotala opened the pull request:
Converted from GDocs to Markdown with pandoc and cleaned up manually.
Tested locally with `npm run serve-docs` as described in docs/workflow.md.

tjotala committed:
New/updated development process documents.

darinkrauss reviewed (approved):
Mostly looks fine. There are some indentation issues with sub-lists. 

If you look here: https://github.com/tidepool-org/docs/blob/new-docs/general/code-peer-review-checklist.md at item 15, you'll note that it contains another list that isn't properly formatted. Other similar issues throughout that document. Also, some issues in https://github.com/tidepool-org/docs/blob/new-docs/general/external-code-dependency-considerations.md.

However, it looks like https://github.com/tidepool-org/docs/blob/new-docs/general/external-service-api-dependency-considerations.md got it right on items 4 and 5 of Security & Privacy Specific Considerations.

Otherwise, LGTM, :shipit:

tjotala commented:
> Mostly looks fine. There are some indentation issues with sub-lists.
> 
> If you look here: https://github.com/tidepool-org/docs/blob/new-docs/general/code-peer-review-checklist.md at item 15, you'll note that it contains another list that isn't properly formatted. Other similar issues throughout that document. Also, some issues in https://github.com/tidepool-org/docs/blob/new-docs/general/external-code-dependency-considerations.md.
> 
> However, it looks like https://github.com/tidepool-org/docs/blob/new-docs/general/external-service-api-dependency-considerations.md got it right on items 4 and 5 of Security & Privacy Specific Considerations.
> 
> Otherwise, LGTM, :shipit:

Oh, that's interesting. The formatting looked OK when I previewed it in Atom's "Markdown Preview" package during editing, and also in pre-flighting locally with `npm run serve-docs`.

Looks like some Markdown renderers (like VSCode) require 4 space indents for nested lists while others (like Atom) are OK with just 2 spaces which is what's in the markdown. I'll update the PR.

tjotala committed:
Fixed indentation to 4 (or 8) spaces, from 2 (or 4).
Some renderers like Atom Markdown Preview accept the smaller indents while VSCode and GitHub do not.

darinkrauss reviewed (approved):
LGTM.

For next steps, I think we might want to boil down the Code Peer Review Checklist to a +/- 10 item list. I think the current list, while pretty exhaustive, is also too large to keep in your head while performing a peer review. We can throw that out to the larger engineering team.
